src/gui/issueswidget.cpp

```diff
@@ -200,4 +261,72 @@ void IssuesWidget::slotItemContextMenu()
     ProtocolWidget::showContextMenu(this, _model, rows);
 }
 
+void IssuesWidget::addStatusFilter(QMenu *menu)
+{
+    menu->addAction(QCoreApplication::translate("OCC::Models", "Status Filter:"))->setEnabled(false);
+
+    // Use a QActionGroup to contain all status filter items, so we can find them back easily to reset.
+    auto statusFilterGroup = new QActionGroup(menu);
+    statusFilterGroup->setExclusive(false);
+
+    const auto initialFilter = _statusSortModel->filter();
+
+    { // Add all errors under 1 action:
+        const std::array<SyncFileItem::Status, 5> ErrorStatusItems = {
+            SyncFileItem::Status::FatalError,
+            SyncFileItem::Status::NormalError,
+            SyncFileItem::Status::SoftError,
+            SyncFileItem::Status::DetailError
+        };
+
+        auto action = menu->addAction(SyncFileItem::statusEnumDisplayName(SyncFileItem::NormalError), [this, ErrorStatusItems](bool checked) {
+            auto currentFilter = _statusSortModel->filter();
+            for (const auto &item : ErrorStatusItems) {
+                currentFilter[item] = checked;
+            }
+            _statusSortModel->setFilter(currentFilter);
+        });
+        action->setCheckable(true);
+        action->setChecked(initialFilter[ErrorStatusItems[0]]);
+        statusFilterGroup->addAction(action);
+    }
+
+    // Add the other non-error items:
+    const std::array<SyncFileItem::Status, 6> OtherStatusItems = {
+        SyncFileItem::Status::Conflict,
+        SyncFileItem::Status::FileIgnored,
+        SyncFileItem::Status::Restoration,
+        SyncFileItem::Status::BlacklistedError,
+        SyncFileItem::Status::Excluded,
+        SyncFileItem::Status::Success
+    };
+    for (const auto &item : OtherStatusItems) {
+        auto action = menu->addAction(SyncFileItem::statusEnumDisplayName(item), [this, item](bool checked) {
+            auto currentFilter = _statusSortModel->filter();
+            currentFilter[item] = checked;
+            _statusSortModel->setFilter(currentFilter);
+        });
+        action->setCheckable(true);
+        action->setChecked(initialFilter[item]);
+        statusFilterGroup->addAction(action);
+    }
+
+    menu->addSeparator();
+
+    // Add action to reset all filters at once:
+    menu->addAction(QCoreApplication::translate("OCC::Models", "Reset Filters"), [statusFilterGroup, this, menu]() {
+        for (QAction *action : statusFilterGroup->actions()) {
+            action->setChecked(true);
+        }
+        _statusSortModel->resetFilter();
+
+        for (QAction *action : menu->actions()) {
+            if (action->objectName() == Models::NoFilterObjectName) {
```

Comment:
Not optimal but I can't come up by something better atm, can you add a todo?

also
````
QObject::findChild
````